spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -79,9 +80,18 @@ public GcpPubSubAutoConfiguration(GcpPubSubProperties gcpPubSubProperties,
 		this.finalProjectIdProvider = gcpPubSubProperties.getProjectId() != null
 				? gcpPubSubProperties::getProjectId
 				: gcpProjectIdProvider;
-		this.finalCredentialsProvider = gcpPubSubProperties.getCredentials().hasKey()
-				? new DefaultCredentialsProvider(gcpPubSubProperties)
-				: credentialsProvider;
+
+		if (gcpPubSubProperties.getEmulatorHost() == null || "false".equals(gcpPubSubProperties.getEmulatorHost())) {
```

Comment:
I guess `!"true".equals(gcpPubSubProperties.getEmulatorHost())` is much better for us.

Reply:
Since the property refers to the emulator host, I cannot check for `!true` here. The check for `false` is the only way to override an already set emulator-host, or is there a better way? All other issues has already been addressed.

Reply:
Oh! I got it.
I think I wound't have any comments here if the story would about a separate `enableEmulator` property.
Although that's just about testing, so I'm fine if you have document that to disable it you need to specify `false` for the `emulatorHost` property. A bit awkward, but let it be if that is what you have agreed here with others...